### PR TITLE
chore: reorder docs and rename Class based workflow to Model based workflow

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**

### CODE REVIEW

**Review by iakovn — CHANGES_REQUESTED** [review]
Thanks for starting this! 🍑 

Please also rename "Class name" into "Model class name" in:
- `modelon/impact/client/entities/model.py`
- `modelon/impact/client/entities/interfaces/model.py`

**Comment by abhilash-kumar-nair** [comment]
> Thanks for starting this! 🍑
> 
> Please also rename "Class name" into "Model class name" in:
> 
> * `modelon/impact/client/entities/model.py`
> * `modelon/impact/client/entities/interfaces/model.py`

Good catch! Fixed it! Thanks. :)

File: docs/source/quick_start.rst [hunk 1]
```diff
@@ -18,27 +18,25 @@ A simple example using the :ref:`FMU based workflow <FMU based workflow>` is sho
    # Choose analysis type
    dynamic = workspace.get_custom_function('dynamic')
 
-   # Compile model
+   # Get model class
```
Comment (iakovn) [inline docs/source/quick_start.rst:21]: Get model reference. ("class" may be confusing)

File: modelon/impact/client/entities/interfaces/model.py [hunk 2]
```diff
@@ -5,5 +5,5 @@ class ModelInterface(ABC):
     @property
     @abstractmethod
     def name(self) -> str:
-        """Class name."""
+        """Model class name."""
```
Comment (iakovn) [inline modelon/impact/client/entities/interfaces/model.py:8]: Sorry for confusion, drop the "class", just "Model name."

File: modelon/impact/client/entities/model.py [hunk 3]
```diff
@@ -86,14 +86,14 @@ def __init__(
...
-        return f"Class name '{self._class_name}'"
+        return f"Model class name '{self._class_name}'"
 
     def __eq__(self, obj: object) -> bool:
         return isinstance(obj, Model) and obj._class_name == self._class_name
 
     @property
     def name(self) -> str:
-        """Class name."""
+        """Model class name."""
```
Comment (iakovn) [inline modelon/impact/client/entities/model.py:96]: Model name.

**Review by iakovn — CHANGES_REQUESTED** [review]
Minor tweaking and please update the tests.

**Comment by abhilash-kumar-nair** [comment]
> Minor tweaking and please update the tests.

Fixed it now! Thanks!

File: modelon/impact/client/entities/experiment.py [hunk 4]
```diff
@@ -541,7 +541,7 @@ def _get_workflow(self) -> _Workflow:
         return _Workflow.CLASS_BASED if model.get("modelica") else _Workflow.FMU_BASED
 
     def get_class_name(self) -> str:
-        """Return the model class name."""
+        """Return the model name."""
```
Comment (iakovn) [inline modelon/impact/client/entities/experiment.py:544]: Given the name of the method the docstring would be unexpected (while `model class name` is ok-ish) . Shall we **add** `get_model_name` and eventually deprecate `get_class_name` (keep for awhile anyway)?

File: modelon/impact/client/entities/model_executable.py [hunk 5]
```diff
@@ -253,7 +253,7 @@ def input(self) -> ModelExecutableInput:
         return ModelExecutableInput(self._get_info()["input"])
 
     def get_class_name(self) -> str:
-        """Return the model class name."""
+        """Return the model name."""
```
Comment (iakovn) [inline modelon/impact/client/entities/model_executable.py:256]: same

**Review by iakovn — CHANGES_REQUESTED** [review]
Unsure about `get_class_name` vs `get_model_name` for the method name.

**Comment by abhilash-kumar-nair** [comment]
> Unsure about `get_class_name` vs `get_model_name` for the method name.

Fixed it! Thanks!

File: modelon/impact/client/entities/model_executable.py [hunk 6]
```diff
@@ -123,7 +123,7 @@ def __init__(self, data: Dict[str, Any]):
 
     @property
     def class_name(self) -> str:
-        """Returns the model class name."""
+        """Returns the model name."""
```
Comment (iakovn) [inline modelon/impact/client/entities/model_executable.py:126]: Deprecate and add "model_name" property.

File: modelon/impact/client/entities/model_executable.py [hunk 7]
```diff
@@ -253,7 +253,12 @@ def input(self) -> ModelExecutableInput:
         return ModelExecutableInput(self._get_info()["input"])
 
     def get_class_name(self) -> str:
-        """Return the model class name."""
+        """Deprecated, use 'get_model_name' attribute."""
```
Comment (iakovn) [inline modelon/impact/client/entities/model_executable.py:256]: """Deprecated, use 'get_model_name' method."""

File: modelon/impact/client/entities/model_executable.py [hunk 8]
```diff
@@ -253,7 +253,12 @@ def input(self) -> ModelExecutableInput:
...
+        logger.warning("This attribute is deprectated, use 'get_model_name' instead!")
```
Comment (iakovn) [inline modelon/impact/client/entities/model_executable.py:257]: How about :
```
import warnings
@warnings.deprecated("The 'get_class_name' is deprecated, use 'get_model_name'")
def get_class_name(self) -> str:
```
if you think this is too much then please fix the spelling of deprec**t**ated

**Review by iakovn — CHANGES_REQUESTED** [review]
some more polishing

**Comment by abhilash-kumar-nair** [comment]
> some more polishing

Good catch! :microscope: ! I have updated the code now.  there is no @warnings.deprecated unfortunately. I have switched to warnings.warn instead

**Review by iakovn — APPROVED** [review]
Looking good! ✈️

**Pull request merged**